pkg/controller/certificates/certificate_controller.go

```diff
@@ -79,29 +79,29 @@ func NewCertificateController(
 	csrInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			csr := obj.(*certificates.CertificateSigningRequest)
-			klog.V(4).Infof("Adding certificate request %s", csr.Name)
+			klog.V(4).InfoS("Adding certificate request", "CSR", klog.KObj(csr))
 			cc.enqueueCertificateRequest(obj)
 		},
 		UpdateFunc: func(old, new interface{}) {
 			oldCSR := old.(*certificates.CertificateSigningRequest)
-			klog.V(4).Infof("Updating certificate request %s", oldCSR.Name)
+			klog.V(4).InfoS("Updating certificate request", "CSR", klog.KObj(oldCSR))
 			cc.enqueueCertificateRequest(new)
 		},
 		DeleteFunc: func(obj interface{}) {
 			csr, ok := obj.(*certificates.CertificateSigningRequest)
 			if !ok {
 				tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
 				if !ok {
-					klog.V(2).Infof("Couldn't get object from tombstone %#v", obj)
+					klog.V(2).InfoS("Couldn't get object from tombstone", "obj", obj)
```

Comment:
The # automatically adds the type information in the old logged output. https://play.golang.org/p/ULYQ34hInsK shows the difference between %#v and %v and how that additional field could be output for the new structured log. This seems important as we are dealing with a cast problem.

Reply:
@cheftako should I revert it to what it was originally or is there a better way to handle this under structured logging?

Reply:
I wouldn't revert for this but a follow on PR to something like 
```
klog.V(2).InfoS("Couldn't get object from tombstone", "obj", obj, "objType", reflect.TypeOf(obj))
```
That way we have only lost the field names. 
Wonder if it would be worth having a helper function in klog for handling the `%#v` case.

Reply:
The alternate would be
```
klog.V(2).InfoS("Couldn't get object from tombstone", "obj", fmt.Sprintf("%#v", obj))
```
Would be nice if we go this route if it had a helper.
This then even gets the field names.